osu.Game/Overlays/SkinEditor/SkinEditor.cs

```diff
@@ -274,6 +282,61 @@ protected override void LoadComplete()
             selectedTarget.BindValueChanged(targetChanged, true);
         }
 
+        private ExternalEditOperation<SkinInfo>? externalEditOperation;
+
+        private async Task editExternally()
+        {
+            mountMenuItem.Action.Disabled = true;
+            var skin = currentSkin.Value.SkinInfo.PerformRead(s => s.Detach());
+
+            try
+            {
+                externalEditOperation = await skins.BeginExternalEditing(skin).ConfigureAwait(false);
+            }
+            catch (Exception ex)
+            {
+                Logger.Log($"Failed to initialize external edit operation: {ex}", LoggingTarget.Database, LogLevel.Error);
+            }
+
+            if (externalEditOperation == null)
+                return;
+
+            host.OpenFileExternally(externalEditOperation.MountedPath.TrimDirectorySeparator() + Path.DirectorySeparatorChar);
+
+            Schedule(() =>
+            {
+                mountMenuItem.Action.Disabled = false;
+                mountMenuItem.Text.Value = EditorStrings.FinishEditingExternally;
+                mountMenuItem.Action.Value = () => _ = finishExternalEdit();
+            });
+        }
+
+        private async Task finishExternalEdit()
+        {
+            if (externalEditOperation == null || !externalEditOperation.IsMounted)
+                return;
+
+            mountMenuItem.Action.Disabled = true;
+
+            await externalEditOperation.Finish().ConfigureAwait(false);
+
+            Schedule(() =>
+            {
+                var oldSkin = currentSkin.Value;
+                var newSkinInfo = oldSkin.SkinInfo.PerformRead(s => s);
+
+                // Create a new skin instance to ensure the skin is reloaded
+                // If there's a better way to reload the skin, this should be replaced with it.
+                currentSkin.Value = newSkinInfo.CreateInstance(skins);
```

Comment:
This is where most of my confidence in the code quality of this PR declines. This was the only way I found that would refresh the skin instantly. I looked through at least 15 different classes, and even tests, to figure out a proper way of reloading skins. Was left empty handed and came up with this which works. The reason I feel unconfident about it is because it feels rather hacky.